source/slang/slang-options.cpp

```diff
@@ -1457,6 +1460,10 @@ struct OptionsParser
             // point was handled above, and the user is expected to
             // follow the ordering rules when using multiple entry points.
             //
+            // TODO(DG): check if output format is left unassociated
+            // If (known format for that output && that format supports whole program) {
+            //    set isWholeProgram bit on the TargetRequest }
+            // else { diagnose what happened }
```

Comment:
Just to note: I think it will be okay to check this PR in with these `TODO` comments in place, if we can reach a good checkpoint. I would just like to make sure that we address all the open `TODO`s related to this feature in follow-on changes.